packages/mui-base/src/SelectUnstyled/SelectUnstyled.types.ts

```diff
@@ -81,10 +86,23 @@ export interface SelectUnstyledOwnProps<TValue extends {}> extends SelectUnstyle
    * The default selected value. Use when the component is not controlled.
    */
   defaultValue?: TValue | null;
+  /**
+   * A function to convert the currently selected value to a string.
+   * Used to set a value of a hidden input associated with the select,
+   * so that the selected value can be posted with a form.
+   */
+  formValueProvider?: (option: SelectOption<TValue> | null) => string | number;
   /**
    * Callback fired when an option is selected.
    */
   onChange?: (value: TValue | null) => void;
+  /**
+   * A function used to convert the option value to a string.
+   * It is used to navigate through values with keyboard.
```

Comment:
I don't really understand from the description what the prop is used for and how it is different than `formValueProvider`